applications/tari_merge_mining_proxy/src/main.rs

```diff
@@ -0,0 +1,704 @@
+
+    // Decode and dserialize the blocktemplate_blob
+    let block_template_blob = &parsed["result"]["blocktemplate_blob"];
+    let s = format!("{}", block_template_blob);
+    let hex =
+        hex::decode(s).map_err(|e| MmProxyError::ParseError(format!("Failure to decode hex {}", e.to_string())))?;
+    let block = deserialize::<Block>(&hex[..])
+        .map_err(|_| MmProxyError::ParseError(format!("Failure to deserialze block ")))?;
+    parsed["result"]["blocktemplate_blob"] = append_merge_mining_tag(&block, Hash(from_slice(hash)))?.into();
+
+    let count = 1 + block.tx_hashes.len() as u16;
+    let mut hashes = block.clone().tx_hashes;
+    hashes.push(block.miner_tx.hash());
+    parsed["result"]["blockhashing_blob"] = create_input_blob(&block.header, &count, &from_hashes(&hashes))?.into();
+    Ok(parsed.dump().into())
+}
+
+// Handles connection from xmrig, passing it through to monerod and back
+#[allow(clippy::unused_io_amount)]
+fn handle_connection(
+    mut stream: TcpStream,
+    mut transient: &mut TransientData,
+    consensus: ConsensusManager,
+    rt: &mut Runtime,
+) -> Result<(), MmProxyError>
+{
+    debug!(target: LOG_TARGET, "Handling Connection");
+    let grpcclientresult = rt.block_on(grpc::base_node_client::BaseNodeClient::connect(TARI_GRPC_URL));
+    let mut buffer = [0; 4096];
+    stream
+        .read(&mut buffer)
+        .map_err(|e| MmProxyError::OtherError(format!("Error reading from stream {}", e.to_string())))?;
+    let date = Local::now();
+    let request_string = stringify_request(&buffer[..]);
+    let request_type = get_request_type(&buffer[..])?;
+    let url_part = get_url_part(&buffer[..])?;
+    if request_type.starts_with("GET") {
+        println!("Handling Method: {}", url_part);
+        // GET requests
+        let url = format!("{}{}", MONEROD_URL, url_part);
+        let mut curl = base_curl(0, &url, false)?;
+        let data = do_curl(&mut curl, b"")?;
+        let response = structure_response(&data[..], 200)?;
+        stream
+            .write(response.as_bytes())
+            .map_err(|e| MmProxyError::OtherError(format!("Error writing to stream {}", e.to_string())))?;
+        stream
+            .flush()
+            .map_err(|e| MmProxyError::OtherError(format!("Error flushing stream {}", e.to_string())))?;
+    } else if request_type.starts_with("POST") {
+        // POST requests
+        let json_bytes = get_json(&buffer[..]);
+        if let Some(json) = json_bytes {
+            let url = format!("{}{}", MONEROD_URL, url_part);
+            let mut curl = base_curl(json.len() as u64, &url, true)?;
+            debug!(target: LOG_TARGET, "Request: {}", request_string);
+            let method = get_post_json_method(&json)?;
+            let mut data = do_curl(&mut curl, &json)?;
+
+            // TODO: Check tari height, monero check height request is type get (/getheight)
+
+            debug!(target: LOG_TARGET, "Handling Method: {}", method.as_str());
```

Comment:
Might want to print this to the console as well as logging it:
```
println!("Handling Method: {}", method.as_str());
```

Or change line 349 to log instead of output, which it currently does here:
```
println!("Handling Method: {}", url_part);
```

Reply:
missed that println!, I specifically dont want to use println. You can now use the log4rs.yml to customizd the output level you want